sdk-cpp/include/do_config.h

```diff
@@ -1,7 +1,10 @@
 #ifndef _DELIVERY_OPTIMIZATION_DO_CONFIG_H
 #define _DELIVERY_OPTIMIZATION_DO_CONFIG_H
 
-#if (DO_CLIENT_ID == DO_CLIENT_ID_AGENT)
+//TODO(jimson): Callers may not have defined these compile definitions, as a result their builds may fail if the definition is not set when using the SDK
+//Look into removing platform specific header files from the SDK installation
+#if defined(DO_CLIENT_AGENT)
```

Comment:
It's tricky because I don't know if we'd ever intend to call these apis for non-agent use cases.

I think this is a potential way forward:
- Modify the sdk cmakelists to only install the right headers (depending on platform & agent), so the .deb file doesn't try to install do_download_property.h for now.
Otherwise, we can implement do_download_property for non-com use cases (will need to do this soon for mac anyways)
- In addition to the above, leave it up to callers who are consuming as source to add these definitions to their builds.

Reply:
Agreed that a compile def that is protecting an  entire .h/.cpp file should be handled via the build system instead. I'll leave that you for later.

Reply:
Also agree that do_download_property should be implemented for all platforms/clients.